x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/transforms/pivot/CompositeBucketsChangeCollector.java

```diff
@@ -383,7 +384,7 @@ public QueryBuilder filterByChanges(long lastCheckpointTimestamp, long nextcheck
                 return null;
             }
             // we only need to round the lower bound, because the checkpoint will not contain new data for the upper bound
```

Comment:
This comment is not relevant anymore, is it?

Reply:
:+1: I removed it in this place, the comment is now part of`collectChangesFromAggregations`.

This change is a small optimization: I moved rounding to `collectChangesFromAggregations`, which is called _exactly once_ per checkpoint. `filterByChanges` is called _more than once_ for every result page (trivia: due to the way composite aggs work, it's called _at least twice_).